crates/nargo_cli/src/cli/info_cmd.rs

```diff
@@ -23,18 +30,26 @@ pub(crate) fn run<B: Backend>(
     args: InfoCommand,
     config: NargoConfig,
 ) -> Result<(), CliError<B>> {
-    count_opcodes_and_gates_with_path(backend, config.program_dir, &args.compile_options)
+    let workspace = resolve_workspace(&config.program_dir)?;
+
+    for package in &workspace {
+        count_opcodes_and_gates_in_package(backend, package, &args.compile_options)?;
+    }
+
+    Ok(())
 }
 
-fn count_opcodes_and_gates_with_path<B: Backend, P: AsRef<Path>>(
+fn count_opcodes_and_gates_in_package<B: Backend>(
     backend: &B,
-    program_dir: P,
+    package: &Package,
     compile_options: &CompileOptions,
 ) -> Result<(), CliError<B>> {
-    let (compiled_program, _) =
-        compile_circuit(backend, None, program_dir.as_ref(), compile_options)?;
+    let (mut context, crate_id) = prepare_package(package);
+    let compiled_program = compile_circuit(backend, &mut context, crate_id, compile_options)?;
+
     let num_opcodes = compiled_program.circuit.opcodes.len();
 
+    // TODO: Print the package name at the beginning?
```

Comment:
Open question: How do we want to handle printed messages? Should we prefix the package name?

Reply:
Vote Yes

Reply:
Yeah, my thoughts on this command is that it should function as described in #1381 so this matches up with that.

Reply:
Updated everywhere that outputs via stdout to include the package names